plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/parameterInfo/KotlinHighLevelFunctionParameterInfoHandler.kt

```diff
@@ -0,0 +1,451 @@
+            } else {
+                // This is for array get/set calls which don't have KtValueArguments.
+                for (argument in arguments) {
+                    val parameter = argumentMapping[argument] ?: continue
+                    val parameterIndex = parameterToIndex[parameter] ?: continue
+                    if (!usedParameterIndices.add(parameterIndex)) continue
+
+                    val shouldHighlight = argumentIndex == currentArgumentIndex
+                    appendParameter(parameter, shouldHighlight)
+                }
+            }
+
+            for ((parameterIndex, parameter) in valueParameters.withIndex()) {
+                if (parameterIndex !in usedParameterIndices) {
+                    if (argumentIndex != parameterIndex) {
+                        namedMode = true
+                    }
+                    // Highlight the first unused parameter if it is in the correct position
+                    val shouldHighlight = !namedMode && highlightStartOffset == -1
+                    appendParameter(
+                        parameter,
+                        shouldHighlight,
+                        markUsedUnusedParameterBorder = namedMode && highlightStartOffset == -1
+                    )
+                }
+            }
+
+            if (length == 0) {
+                append(CodeInsightBundle.message("parameter.info.no.parameters"))
+            }
+        }
+
+        val backgroundColor = if (isCallResolvedToCandidate) GREEN_BACKGROUND else context.defaultParameterColor
+        // TODO: Strikeout if deprecated. Deprecation status not currently in HL API
+        val strikeout = false
+
+        // Disabled when there are too many arguments.
+        val allParametersUsed = usedParameterIndices.size == valueParameters.size
+        val supportsTrailingCommas = callElement.languageVersionSettings.supportsFeature(LanguageFeature.TrailingCommas)
+        val afterTrailingComma = arguments.isNotEmpty() && currentArgumentIndex == arguments.size
+        val tooManyArgs = allParametersUsed && ((!supportsTrailingCommas && afterTrailingComma) || arguments.size > valueParameters.size)
+
+        // TODO: Also disable if not all arguments before the current are matched. Need to check this in updateParameterInfo()
+        // See UpdateInTyping.kt tests
+        val hasMismatchBeforeCurrent = false
+
+        val isDisabled = tooManyArgs || hasMismatchBeforeCurrent
+
+        context.setupUIComponentPresentation(
+            text,
+            highlightStartOffset,
+            highlightEndOffset,
+            isDisabled,
+            strikeout,
+            isDisabledBeforeHighlight,
+            backgroundColor
+        )
+
+        return true
+    }
+
+    data class CallInfo(
+        val callElement: KtElement,
+        val valueArguments: List<KtValueArgument>?,
+        val arguments: List<KtExpression>,
+        val argumentMapping: LinkedHashMap<KtExpression, KtValueParameterSymbol>,
```

Comment:
KtValueParameterSymbol cannnot be stored in a class, only symbol pointers may be KtSymbolPointer may be stored. It violates the contracts that KtSymbols should not be leaked outside KtAnalysisSessions they were created in

Reply:
Or maybe just store the final rendered strings in here instead?

Reply:
I'm storing the index as values instead. See latest version. I still need `parameterToText` map to render the parameters with no arguments.